src/sentry/workflow_engine/typings/notification_action.py

```diff
@@ -354,15 +395,13 @@ def target_type(self) -> ActionTarget:
             return ActionTarget.TEAM
         return ActionTarget.ISSUE_OWNERS
 
-    @property
-    def integration_id(self) -> None:
-        return None
-
     @property
     def target_identifier(self) -> str | None:
         target_type = self.action.get("targetType")
         if target_type in [ActionTargetType.MEMBER.value, ActionTargetType.TEAM.value]:
-            return self.action.get("targetIdentifier")
+            return self.action.get(
+                ACTION_FIELD_MAPPINGS[Action.Type.EMAIL]["target_identifier_key"]
```

Comment:
nit: Do we want enums for each of these field names now? Might be good just to prevent us from having typo related bugs.